functions/grapher/[slug].ts

```diff
@@ -106,5 +131,69 @@ export const onRequestGet: PagesFunction = async (context) => {
             },
         })
 
-    return rewriter.transform(grapherPageResp)
+    return rewriter.transform(grapherPageResp as unknown as Response)
+}
+
+async function handleConfigRequest(
+    slug: string,
+    searchParams: URLSearchParams,
+    env: Env,
+    etag: string | undefined
+) {
+    const shouldCache = searchParams.get("nocache") === null
+    console.log("Preapring json response for ", slug)
+    // All our grapher slugs are lowercase by convention.
+    // To allow incoming links that may contain uppercase characters to work, we redirect to the lowercase version.
+    const lowerCaseSlug = slug.toLowerCase()
+    if (lowerCaseSlug !== slug) {
+        const redirectSlug = await getOptionalRedirectForSlug(
+            lowerCaseSlug,
+            env.url,
+            env
+        )
+
+        return createRedirectResponse(
+            redirectSlug ? `${redirectSlug}.config.json` : lowerCaseSlug,
+            env.url
+        )
+    }
+
+    const grapherPageResp = await fetchGrapherConfig(slug, env, etag)
+
+    if (grapherPageResp.status === 304) {
+        return new Response(null, { status: 304 })
+    }
+
+    if (grapherPageResp.status !== 200) {
+        // If the request is a 404, we check if there's a redirect for it.
+        // If there is, we redirect to the new page.
+        const redirectSlug = await getOptionalRedirectForSlug(
+            slug,
+            env.url,
+            env
+        )
+        if (redirectSlug && redirectSlug !== slug) {
+            return createRedirectResponse(
+                `${redirectSlug}.config.json`,
+                env.url
+            )
+        } else {
+            // Otherwise we just return the status code.
+            return new Response(null, { status: grapherPageResp.status })
+        }
+    }
+
+    console.log("Grapher page response", grapherPageResp.grapherConfig.title)
+
+    const cacheControl = shouldCache
+        ? "public, s-maxage=3600, max-age=0, must-revalidate"
+        : "public, s-maxage=0, max-age=0, must-revalidate"
+
+    return new Response(JSON.stringify(grapherPageResp.grapherConfig), {
+        headers: {
+            "content-type": "application/json",
```

Comment:
So, for once it's definitely worth it to change this to:
```suggestion
    return Response.json(grapherPageResp.grapherConfig, {
        headers: {
            "content-type": "application/json",
```


But on the other hand I'm also wondering if we need the whole `JSON.parse` and `JSON.serialize` in the first place?

I'm mostly asking because this would allow us to use CF's streaming, where it is sending the response while it is still reading the response from the origin server.